VideoLocker/src/main/java/org/edx/mobile/http/Api.java

```diff
@@ -1223,6 +1221,12 @@ public RegistrationDescription getRegistrationDescription() throws Exception {
     }
 
     public boolean enrollInACourse(String courseId, boolean email_opt_in) throws Exception {
+        // uses auth headers
+        Bundle headers = getAuthHeaders();
+
+        // this API endpoint also requires CSRFToken, so add it
+        headers = setCookieHeaders(headers);
```

Comment:
No, <code>getLoginResponseHeaders()</code> has a network request. However, this method itself is only being called for <code>resetPassword</code>.  Reason was, resetPassword was using <code>CSRFToken</code>. 

<code>getAuthHeaders</code> method does not make any network request.